pallets/collator-assignment/src/lib.rs

```diff
@@ -398,47 +402,43 @@ pub mod pallet {
 
             // We assign new collators
             // we use the config scheduled at the target_session_index
-            let new_assigned =
-                if T::ShouldRotateAllCollators::should_rotate_all_collators(target_session_index) {
-                    log::debug!(
-                        "Collator assignment: rotating collators. Session {:?}, Seed: {:?}",
-                        current_session_index.encode(),
-                        random_seed
-                    );
+            let full_rotation =
+                T::ShouldRotateAllCollators::should_rotate_all_collators(target_session_index);
+            if full_rotation {
+                log::debug!(
+                    "Collator assignment: rotating collators. Session {:?}, Seed: {:?}",
+                    current_session_index.encode(),
+                    random_seed
+                );
+            } else {
+                log::debug!(
+                    "Collator assignment: keep old assigned. Session {:?}, Seed: {:?}",
+                    current_session_index.encode(),
+                    random_seed
+                );
+            }
 
-                    Self::deposit_event(Event::NewPendingAssignment {
-                        random_seed,
-                        full_rotation: true,
-                        target_session: target_session_index,
-                    });
-
-                    Assignment::<T>::assign_collators_rotate_all(
-                        collators,
-                        orchestrator_chain,
-                        chains,
-                        shuffle_collators,
-                    )
-                } else {
-                    log::debug!(
-                        "Collator assignment: keep old assigned. Session {:?}, Seed: {:?}",
-                        current_session_index.encode(),
-                        random_seed
-                    );
+            Self::deposit_event(Event::NewPendingAssignment {
```

Comment:
it might be a breaking change, but I think it would be good to see in the event the rotation type that happened

Reply:
Also it feels a little bit awkward to through the event before all changes happen no? I know we were doing this in the past, but maybe it's a good time to push the event a little be downwards

Reply:
Yeah, I was going to change it and emit the event after, but let's do it in a separate PR

Reply:
The type depends on the chain, do you want the event to contain the full "FullRotationModes" struct?

Reply:
*fixed*